editor.planx.uk/src/pages/Preview/Questions.tsx

```diff
@@ -69,59 +69,62 @@ const Questions = ({ previewEnvironment }: QuestionsProps) => {
   const isStandalone = previewEnvironment === "standalone";
   const { createAnalytics, node } = useAnalyticsTracking();
   const [gotFlow, setGotFlow] = useState(false);
-  const isSingleSession =
+  const isUsingLocalStorage =
     useStore((state) => state.path) === ApplicationPath.SingleSession;
 
-  if (isSingleSession) {
-    // Use local storage for simple, non-Save&Return flows
-    useEffect(() => {
-      setPreviewEnvironment(previewEnvironment);
-      if (isStandalone) {
-        const state = getLocalFlow(id);
-        if (state) {
-          resumeSession(state);
-        }
+  // Initial setup
+  useEffect(() => {
```

Comment:
These are now grouped into 2 `useEffects()` instead of 4 - it feels clearer and simpler to me this way. One for initial setup, one when questions are answered to save the session.